src/Tizen.NUI.Gadget/Tizen.NUI/NUIGadget.cs

```diff
@@ -0,0 +1,265 @@
+﻿/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.ComponentModel;
+using Tizen.Applications;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI
+{
+    /// <summary>
+    /// This class represents a NUIGadget controlled lifecycles.
+    /// </summary>
+    /// <since_tizen> 11 </since_tizen>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class NUIGadget
+    {
+        /// <summary>
+        /// Initializes the gadget.
+        /// </summary>
+        /// /// <param name="type">The type of the NUIGadget.</param>
+        /// <since_tizen> 11 </since_tizen>
+        public NUIGadget(NUIGadgetType type)
+        {
+            Type = type;
+            State = NUIGadgetLifecycleState.Initialized;
+        }
+
+        internal event EventHandler<NUIGadgetLifecycleChangedEventArgs> LifecycleChanged;
+
+        /// <summary>
+        /// Gets the class representing information of the current gadget.
+        /// </summary>
+        /// <since_tizen> 11 </since_tizen>
+        public NUIGadgetInfo NUIGadgetInfo
+        {
+            internal set;
+            get;
+        }
+
+        /// <summary>
+        /// Gets the type.
+        /// </summary>
+        /// <since_tizen> 11 </since_tizen>
+        public NUIGadgetType Type
+        {
+            internal set;
+            get;
+        }
+
+        /// <summary>
+        /// Gets the class name.
+        /// </summary>
+        /// <since_tizen> 11 </since_tizen>
+        public string ClassName
+        {
+            internal set;
+            get;
+        }
+
+        /// <summary>
+        /// Gets the main view.
+        /// </summary>
+        /// <since_tizen> 11 </since_tizen>
+        public View MainView
+        {
+            internal set;
+            get;
+        }
+
+        /// <summary>
+        /// Gets the lifecycle state.
+        /// </summary>
+        /// <since_tizen> 11 </since_tizen>
+        public NUIGadgetLifecycleState State
+        {
+            internal set;
+            get;
+        }
+
+        internal void Create()
+        {
+            MainView = OnCreate();
```

Comment:
I think making view is not an option.
How about changing return type and handling exceptions at caller?
```cs
internal bool Create()
{
    MainView = OnCreate();
    if (MainView == null)
       return false;
    return true;
}

```

Reply:
Done. Thanks.

I've updated the patchset.
If the Create() method returns 'false', the Add() method throws the InvalidOperationException.